lib/internal/fs/streams.js

```diff
@@ -273,7 +273,8 @@ function closeFsStream(stream, cb, err) {
 }
 
 ReadStream.prototype.close = function(cb) {
-  this.destroy(null, cb);
+  finished(this, cb);
```

Comment:
I think we need to handle the case of `cb` being null/undefined.

Reply:
good catch, thanks!